labgrid/remote/client.py

```diff
@@ -696,26 +696,30 @@ def power(self):
         from ..driver.powerdriver import NetworkPowerDriver, PDUDaemonDriver, USBPowerDriver
         from ..resource.power import NetworkPowerPort, PDUDaemonPort
         from ..resource.remote import NetworkUSBPowerPort
+
         drv = None
-        for resource in target.resources:
-            if isinstance(resource, NetworkPowerPort):
-                try:
-                    drv = target.get_driver(NetworkPowerDriver)
-                except NoDriverFoundError:
-                    drv = NetworkPowerDriver(target, name=None, delay=delay)
-                break
-            elif isinstance(resource, NetworkUSBPowerPort):
-                try:
-                    drv = target.get_driver(USBPowerDriver)
-                except NoDriverFoundError:
-                    drv = USBPowerDriver(target, name=None, delay=delay)
-                break
-            elif isinstance(resource, PDUDaemonPort):
-                try:
-                    drv = target.get_driver(PDUDaemonDriver)
-                except NoDriverFoundError:
-                    drv = PDUDaemonDriver(target, name=None, delay=int(delay))
-                break
+        try:
+            target.get_driver("PowerProtocol")
```

Comment:
I assume this should be `` drv = target.get_driver("PowerProtocol")``.
Otherwise you will end up without a Driver, even if a protocol is present.

Reply:
indeed. fixed.